src/main/java/com/kodedu/cloudterm/websocket/TerminalSocket.java

```diff
@@ -73,7 +73,8 @@ public void handleTransportError(WebSocketSession session, Throwable exception)
 
     @Override
     public void afterConnectionClosed(WebSocketSession session, CloseStatus status) throws Exception {
-        super.afterConnectionClosed(session, status);
+        session.close();
```

Comment:
I have just a concern here that onTerminalClose wouldn't be called if close throws an exception. Probably it may happen in rare case but I recommend calling close in try/catch

Reply:
> I have just a concern here that onTerminalClose wouldn't be called if close throws an exception. Probably it may happen in rare case but I recommend calling close in try/catch

I have checked it, the method of `afterConnectionClosed` will be called when client close connection positively, such as browser refresh. In this situation, session status will be [`GOING_AWAY / !State.OPEN`](https://github.com/spring-projects/spring-framework/blob/ab0e6515479f58d2f12b759ee56cc3218f71d2aa/spring-websocket/src/main/java/org/springframework/web/socket/CloseStatus.java#L45). 
In finally, `close` will invoke at [org.apache.tomcat.websocket.WsSession#doClose](https://github.com/wenzhucjy/tomcat_source/blob/2aef64740ef214d382913f12b3c56a3fec36f4fb/tomcat-8.0.9-sourcecode/java/org/apache/tomcat/websocket/WsSession.java#L403) , it will do double check session state which is open.  
```java
    public void doClose(CloseReason closeReasonMessage, CloseReason closeReasonLocal,
            boolean closeSocket) {
        // Double-checked locking. OK because state is volatile
        if (state != State.OPEN) {
            return;
        }

        synchronized (stateLock) {
            if (state != State.OPEN) {
                return;
            }
....
```
So `session.close()` is safe.
As you say, in the rare case, maybe calling close in try/catch is better, I have fixed it.